File: c#/mybatisGen/WindowsFormsApplication1/bin/Debug/GenTarget/20190906102509/Controllers/IRobotUserController.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Web;
using WebApplication1.Entity;
namespace WebApplication1.Controllers
{
	/// <summary>
	/// ***模块的控制器类
	/// </summary>
    public sealed partial class IRobotUserController : BaseController
    {
		public IRobotUserService Service { set; get; }

		/*抄考代码
----------------------------------------------------------------------------------------------------------------------------
		IRobotUser data = new IRobotUser
		{

					IUId = Next(),
		
					IUUsername = param.IUUsername,
		
		
					IUPassword = param.IUPassword,
		
		
					IUSignKey = param.IUSignKey,
		
		
					IUSignSecret = param.IUSignSecret,
		
		
					IUCreateDate = param.IUCreateDate,
		
		
		
		};
----------------------------------------------------------------------------------------------------------------------------
		IRobotUserParams param = new IRobotUserParams
		{

			IUId = param.IUId,
		
			IUUsername = param.IUUsername,
		
			IUUsernameLike = param.IUUsernameLike,
		
			IUPassword = param.IUPassword,
		
			IUPasswordLike = param.IUPasswordLike,
		
			IUSignKey = param.IUSignKey,
		
			IUSignKeyLike = param.IUSignKeyLike,
		
			IUSignSecret = param.IUSignSecret,
		
			IUSignSecretLike = param.IUSignSecretLike,
		
			IUCreateDate = param.IUCreateDate,
		
			IUCreateDateStart = param.IUCreateDateStart,
		
			IUCreateDateEnd = param.IUCreateDateEnd,
		
		};
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 分页查询***模块，并返回查询结果
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的查询结果</returns>
		[Compress]
		public JsonResult Page(IRobotUserParams param,int currentPageIndex = 1,int pageSize = 20)
		{
			return MyJson(new Result{code = 0, data = Service.Page(param, currentPageIndex, pageSize)});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 导出***模块，使用excel表保存
        /// </summary>
        /// <param name="param">查询参数</param>
		/// <param name="excelType">excel类型</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的导出结果</returns>
		[OperInterval(IntervalMillisecond=10000)]
		[Compress]
		public ExcelResult<IRobotUser> Export(IRobotUserParams param, string excelType, int currentPageIndex = 1, int pageSize = 10000)
		{
			return new ExcelResult<IRobotUser>
			{
				DataList = Service.Page(param, currentPageIndex, pageSize).pageDataList,
				FileName = "测试excel."+excelType
			};
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 导入***模块数据
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的导出结果</returns>
		public JsonResult Import(HttpPostedFileBase fileUpload, 其他参数...)
		{
			List<IRobotUser> IRobotUserList;
            if (fileUpload.FileName.EndsWith("xlsx"))
            {
                IRobotUserList = ExcelHelper.ExcelXlsxToList<IRobotUser>(fileUpload.InputStream);
            }
            else if (fileUpload.FileName.EndsWith("xls"))
            {
                IRobotUserList = ExcelHelper.ExcelXlsToList<IRobotUser>(fileUpload.InputStream);
            }
			Service.AddBatch(IRobotUserList);
			return MyJson(new Result { code = 0, msg = "导入成功。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 根据主键删除指定数据
        /// </summary>
        /// <param name="IUId">删除数据的主键</param>
		public JsonResult Del(long IUId)
		{
            Service.Del(IUId);
			return MyJson(new Result { code = 0, msg = "数据已删除。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 新增一条数据
        /// </summary>
        /// <param name="data">新增的数据</param>
		public JsonResult Add(IRobotUser data)
		{
			Service.Add(data);
			return MyJson(new Result { code = 0, msg = "保存成功。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 批量修改状态
        /// </summary>
        /// <param name="datas">修改状态的数据</param>
		public JsonResult ChangeStatus(IRobotUser datas)
		{
			return MyJson(new Result { code = 0,msg=$"修改成功，共{Service.ChangeStatus(datas)}条。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 根据主键id查询***模块的数据实体
        /// </summary>
        /// <param name="IUId">主键id</param>
		/// <returns>返回***模块的查询结果</returns>
		public JsonResult Load(long IUId)
		{
			return MyJson(new Result { code = 0, data = Service.Load(IUId) });
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 批量删除数据
        /// </summary>
        /// <param name="datas">批量删除的数据</param>
		public JsonResult DelBatch(List<IRobotUser> datas)
		{
			return MyJson(new Result { code = 0 ,msg = $"删除成功，共{Service.DelBatch(datas)}条。" });
		}
		*/
    }
}

File: java/web/src/main/java/web/template/controller/IRobotUserController.java
package web.template.controller;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import web.template.controller.common.BaseController;
import web.template.service.IRobotUserService;
/**
 * “用户管理”模块的控制器类
 * @author admin
 */
@RestController
@RequestMapping("/IRobotUser")
public class IRobotUserController extends BaseController{
	
	@Autowired
	private IRobotUserService service;
	
	/*抄考代码
	----------------------------------------------------------------------------------------------------------------------------
	IRobotUser data = new IRobotUser();

					data.setIuId(nextId());
		
					data.setIuUsername(param.getIuUsername());
		
		
					data.setIuPassword(param.getIuPassword());
		
		
					data.setIuSignKey(param.getIuSignKey());
		
		
					data.setIuSignSecret(param.getIuSignSecret());
		
		
					data.setIuCreateDate(param.getIuCreateDate());
		
		
		
	----------------------------------------------------------------------------------------------------------------------------
	IRobotUserParams param = new IRobotUserParams();

		param.setIuId(param.getIuId());
		
		param.setIuUsername(param.getIuUsername());
		
		param.setIuUsernameLike(param.getIuUsernameLike());
		
		param.setIuPassword(param.getIuPassword());
		
		param.setIuPasswordLike(param.getIuPasswordLike());
		
		param.setIuSignKey(param.getIuSignKey());
		
		param.setIuSignKeyLike(param.getIuSignKeyLike());
		
		param.setIuSignSecret(param.getIuSignSecret());
		
		param.setIuSignSecretLike(param.getIuSignSecretLike());
		
		param.setIuCreateDate(param.getIuCreateDate());
		
		param.setIuCreateDateStart(param.getIuCreateDateStart());
		
		param.setIuCreateDateEnd(param.getIuCreateDateEnd());
		
	----------------------------------------------------------------------------------------------------------------------------
	* 分页查询***模块，并返回查询结果
	* @param param	查询参数
    * @param currentPageIndex	当前页码
 	* @param pageSize	每页显示的数据量
    * @return	返回***模块的查询结果
    @RequestMapping("/page")
	public Result page(IRobotUserParams param,int currentPageIndex,int pageSize){
		return new Result(0,null,service.page(param, currentPageIndex, pageSize));
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 导出***模块，使用excel表保存
	* @param param 筛选参数
	* @param excelType excel的后缀名、类型
	* @param currentPageIndex 当前页码
	* @param pageSize 每页显示的数据量
	* @return 返回excel
	@RequestMapping("/export")
	public ModelAndView export(IRobotUserParams param, String excelType, int currentPageIndex, int pageSize){
		Map<String,Object> model=new HashMap<String,Object>();
		model.put(MyExcelView.EXCEL_DATA, service.export(param, currentPageIndex, pageSize).getPageDataList());
		model.put(MyExcelView.EXCEL_NAME, "excel文件名称."+excelType);
		model.put(MyExcelView.GROUP_NAME, "组名称");
		return new ModelAndView(myExcelView, model);
	}
	----------------------------------------------------------------------------------------------------------------------------
	@RequestMapping("/import")
	public Result importExcel(@RequestParam("fileUpload")MultipartFile fileUpload, 其他参数...){
		List<IRobotUser> IRobotUserList;
        if (fileUpload.getOriginalFilename().endsWith("xlsx"))
        {
            IRobotUserList = ExcelHelper.excelXlsxToList<IRobotUser>(fileUpload.InputStream);
        }
        else if (fileUpload.getOriginalFilename().endsWith("xls"))
        {
            IRobotUserList = ExcelHelper.excelXlsToList<IRobotUser>(fileUpload.InputStream);
        }
		service.addBatch(IRobotUserList);
		return new Result(0, "导入成功。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	@RequestMapping("/del")
	public Result del(long iuId){
        service.del(iuId);
		return new Result(0, "数据已删除。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 新增一条数据
	* @param data	新增的数据
	@RequestMapping("/add")
	public Result add(IRobotUser data){
		service.add(data);
		return new Result(0, "保存成功。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 批量修改状态
	* @param datas	  修改状态的数据
	@RequestMapping("/changeStatus")
	public Result changeStatus(IRobotUser datas){
		return new Result(0,"修改成功，共"+service.changeStatus(datas)+"条。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 根据主键id查询***模块的数据实体
	* @param iuId	主键id
	* @param return 返回***模块的查询结果
	@RequestMapping("/load")
	public Result load(Long iuId){
		return new Result(0,null ,service.Load(iuId) );
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 批量删除***模块的数据实体
	* @param datas	删除的数据主键列表
	@RequestMapping("/delBatch")
	public Result delBatch(List<IRobotUser> datas){
		return new Result(0 ,null,"删除成功，共"+Service.DelBatch(datas)+"条。");
	}
	 */
}